pkg/command/plugin_group.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package command
+
+import (
+	"fmt"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/pluginmanager"
+
+	"github.com/spf13/cobra"
+
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/component"
+)
+
+func newPluginGroupCmd() *cobra.Command {
+	var pluginGroupCmd = &cobra.Command{
+		Use:   "group",
+		Short: "Manage plugin groups",
+		Long:  "Manage plugin groups. A plugin group provides a list of plugins and version combinations, which can be installed in one step.",
+	}
+	pluginGroupCmd.SetUsageFunc(cli.SubCmdUsageFunc)
+
+	pluginGroupCmd.AddCommand(
+		newSearchCmd(),
+	)
+
+	return pluginGroupCmd
+}
+
+func newSearchCmd() *cobra.Command {
+	var searchCmd = &cobra.Command{
+		Use:   "search",
+		Short: "Search for available plugin groups",
+		Long:  "Search from the list of available plugin groups.  A plugin group provides a list of plugins and version combinations, which can be installed in one step.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			output := component.NewOutputWriter(cmd.OutOrStdout(), outputFormat, "group", "source")
+
+			groupsByDiscovery, err := pluginmanager.DiscoverPluginGroups()
+			if err != nil {
+				return err
+			}
+			for _, discAndGroups := range groupsByDiscovery {
+				for _, group := range discAndGroups.Groups {
+					output.AddRow(fmt.Sprintf("%s/%s", group.Publisher, group.Name), discAndGroups.Source)
```

Comment:
nit: we don't display Source in plugin search results, probably because it is not that significant for plugins.
Are we anticipating possible conflict and that it is actually significant if there are identically named groups which one to use? 
Any way, pretty harmless imo to show the source, so we can have followup discussions without blocking this.

Reply:
I am not sure I understand the point here but I would like to be part of the discussion.

Reply:
This is a good observation.  I added it while doing the implementation because in the case of a name conflict (in the code we have to protect against it, even if it should not happen) I wanted to allow the user to know from which discovery source the conflict was caused.  A far-fetched case that can be used as an example is, if the user adds the same discovery source twice by mistake.

But you're right that this extra information is going to be useless for the vast majority of situations. What I'll do instead is print a error message if multiple plugins groups are found with the same name.

Note that for plugins, we combine the plugins that were found with the same name and target. I don't think we should do this for plugin groups or at least not without thinking about it in more detail.

Reply:
This is what the warning message looks like:
```
$ rm ~/.config/tanzu/config*
$ tz config set features.global.central-repository true
$ tz plugin clean
✔  successfully cleaned up all plugins

# Add the same source twice to get duplicate groups
$ export TANZU_CLI_PRE_RELEASE_REPO_IMAGE=TEST_BYPASS
$ tz plugin source add -n default -t oci -u localhost:9876/tanzu-cli/plugins/central:small
✔  successfully added discovery source default
$ tz plugin source add -n default2 -t oci -u localhost:9876/tanzu-cli/plugins/central:small
✔  successfully added discovery source default2

$ tz plugin group search
!  vmware-tkg/tkg-v2.1.0 was found in more than one source: [default default2], vmware-tmc/tmc-v1.2.3 was found in more than one source: [default default2], vmware-tmc/tmc-v9.0.0 was found in more than one source: [default default2], vmware-tkg/tkg-v1.0.0 was found in more than one source: [default default2]
  GROUP
  vmware-tkg/tkg-v1.0.0
  vmware-tkg/tkg-v2.1.0
  vmware-tmc/tmc-v1.2.3
  vmware-tmc/tmc-v9.0.0
  vmware-tkg/tkg-v1.0.0
  vmware-tkg/tkg-v2.1.0
  vmware-tmc/tmc-v1.2.3
  vmware-tmc/tmc-v9.0.0
```